### PR TITLE
Added tests - PUB-13279 - also updated transform Advanced to better d…

### PR DESCRIPTION
**Pull request by divanmostert**
…eal with simple inputs

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
         <jdk.version>1.8</jdk.version>
         <category>Community</category>
         <licensePath>LICENSE.md</licensePath>
-        <c24.api.version>4.11.4</c24.api.version>
+        <c24.api.version>4.10.0</c24.api.version>
```
Comment (dedwards60) [inline pom.xml]: Is this version going backwards what was intended

File: src/test/java/BasicOperationsTestCase.java [hunk 2]
```diff
@@ -14,20 +14,109 @@
...
+        File testFile = new File("./src/test/resources/testdata/InvalidCustomers.xml");
+        byte[] contents = Files.readAllBytes(testFile.toPath());
+
+        Object error = null;
+        try {
+            error = flowRunner("validate")
+                    .withPayload(new String(contents))
+                    .run().getError();
+        } catch (Exception e) {
+            //TODO: handle properly
```
Comment (dedwards60) [inline src/test/java/BasicOperationsTestCase.java]: probably want to remove the TODO

File: pom.xml (hunk 1 above)
Comment (divanmostert) [reply pom.xml]: Yes, the current generated library that's already associated with the project did not work with 4.11

File: src/test/java/BasicOperationsTestCase.java (hunk 2 above)
Comment (divanmostert) [reply src/test/java/BasicOperationsTestCase.java]: Yes, good point

**Pull request merged**